plugins/woocommerce/src/Internal/Admin/Settings/PaymentsRestController.php

```diff
@@ -0,0 +1,1094 @@
+			'description' => esc_html__( 'Offer flexible payment options to your shoppers.', 'woocommerce' ),
+		);
+		$categories[] = array(
+			'id'          => 'psp',
+			'_priority'   => 30,
+			'title'       => esc_html__( 'Payment Providers', 'woocommerce' ),
+			'description' => esc_html__( 'Give your shoppers additional ways to pay.', 'woocommerce' ),
+		);
+
+		return $categories;
+	}
+
+	/**
+	 * General permissions check for payments settings REST API endpoint.
+	 *
+	 * @param WP_REST_Request $request The request for which the permission is checked.
+	 * @return bool|WP_Error True if the current user has the capability, otherwise an "Unauthorized" error or False if no error is available for the request method.
+	 */
+	private function check_permissions( WP_REST_Request $request ) {
+		$context = 'read';
+		if ( 'POST' === $request->get_method() ) {
+			$context = 'edit';
+		} elseif ( 'DELETE' === $request->get_method() ) {
+			$context = 'delete';
+		}
+
+		if ( wc_rest_check_manager_permissions( 'payment_gateways', $context ) ) {
+			return true;
+		}
+
+		$error_information = $this->get_authentication_error_by_method( $request->get_method() );
+		if ( is_null( $error_information ) ) {
+			return false;
+		}
+
+		return new WP_Error(
+			$error_information['code'],
+			$error_information['message'],
+			array( 'status' => rest_authorization_required_code() )
+		);
+	}
+
+	/**
+	 * Get the accepted arguments for the GET payment providers request.
+	 *
+	 * @return array[] The accepted arguments for the GET request.
+	 */
+	private function get_args_for_get_payment_providers(): array {
+		return array(
+			'location' => array(
+				'description'       => __( 'ISO3166 alpha-2 country code. Defaults to WooCommerce\'s base location country.', 'woocommerce' ),
+				'type'              => 'string',
+				'required'          => false,
+				'sanitize_callback' => 'sanitize_text_field',
+			),
+		);
+	}
+
+	/**
+	 * Get the schema for the GET payment providers request.
+	 *
+	 * @return array[]
+	 */
+	private function get_schema_for_get_payment_providers(): array {
+		$schema               = array(
+			'$schema' => 'http://json-schema.org/draft-04/schema#',
+			'title'   => 'WooCommerce Settings Payments providers for the given location.',
+			'type'    => 'object',
+		);
+		$schema['properties'] = array(
+			'gateways'                => array(
+				'type'        => 'array',
+				'description' => esc_html__( 'The registered payment gateways.', 'woocommerce' ),
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+				'items'       => $this->get_schema_for_payment_gateway(),
+			),
+			'offline_payment_methods' => array(
+				'type'        => 'array',
+				'description' => esc_html__( 'The offline payment methods.', 'woocommerce' ),
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+				'items'       => $this->get_schema_for_payment_gateway(),
```

Comment:
offline payment methods has its own schema, it's not equal to registered PGs.
it doesn't contain image, icon, links, tags, plugin.
I believe that it should contain image and icon - WDYT @vladolaru ?

Reply:
It seems [there is a way](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/) to specify required properties (since, by default, all are optional).

Reply:
But let's leave this fine tuning for a later time. This is not an endpoint for public consumption, it is only for us (internal).

Reply:
@oaratovskyi I added icons for offline PMs in https://github.com/woocommerce/woocommerce/pull/52647/commits/c1d761f153d7293b9393dbc70d1ceab56979330d